Makefile

```diff
@@ -4,9 +4,12 @@ IMAGE_VERSION?=$(shell git tag | tail -1)
 IMAGE_TAG=$(REGISTRY_NAME)/$(IMAGE_NAME):$(IMAGE_VERSION)
 IMAGE_TAG_LATEST=$(REGISTRY_NAME)/$(IMAGE_NAME):latest
 BUILD_DATE=$$(date +%Y-%m-%d-%H:%M)
-LDFLAGS?="-X github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/version.BuildVersion=$(IMAGE_VERSION) -X github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/version.BuildDate=$(BUILD_DATE) -extldflags "-static""
+LDFLAGS?="-X 'github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/version.BuildVersion=$(IMAGE_VERSION)' \
+	-X 'github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/version.BuildDate=$(BUILD_DATE)' \
+	-X 'github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/version.GoVersion=$(shell go version)' \
```

Comment:
Will this work on windows (the `shell` command)? Would it be better to set this version as a variable like we do for `BUILD_DATE`? Granted I see now that we do `shell` for `IMAGE_VERSION` as well

Reply:
Thanks for pointing this out, I hadn't really considered that. I don't have a Windows machine to test on, but I think it may work? As `shell` is just a Makefile function that says "use the default shell" and I believe it will use `cmd.exe` and I _think_ `go version` will work? There are a lot of guesses there, but as mentioned it doesn't change compatibility and Windows support is a fair way off for now, so I'll leave this one as-is for this PR.

[Makefile shell docs for reference](https://www.gnu.org/software/make/manual/html_node/Shell-Function.html)
[Choosing the shell docs](https://www.gnu.org/software/make/manual/html_node/Choosing-the-Shell.html) which make it sound way more complicated than I said above 😱